Add "How to Document Adapter" guide (#1467)
fishtown-analytics/docs.getdbt.com: website/docs/docs/contributing/building-a-new-adapter.md

2. In the middle: A set of *macros* responsible for generating SQL that is compliant with the target database.
3. (Optional) At the highest level: A set of *<Term id="materialization">materializations</Term>* that tell dbt how to turn model files into persisted objects in the database.

> This guide will walk you through the first two steps, and provide some resources to help you validate that your new adapter is working correctly. Once you feel it is mostly working, please let the community know that is available to use by adding the adapter to the [Available Adapters](/website/docs/docs/available-adapters.md) page by following the steps given in [Documenting your adapter](/website/docs/docs/contributing/documenting-a-new-adapter.md).

Reviewer: > Once you feel it is mostly working

Could we be more...specific here? This sounds very confusing.

Author: good call. improved!

2. In the middle: A set of *macros* responsible for generating SQL that is compliant with the target database.
3. (Optional) At the highest level: A set of *<Term id="materialization">materializations</Term>* that tell dbt how to turn model files into persisted objects in the database.

This guide will walk you through the first two steps, and provide some resources to help you validate that your new adapter is working correctly. Once the adapter is passing most of the functional tests (see ["Testing a new adapter"](testing-a-new-adapter)

> ), please let the community know that is available to use by adding the adapter to the [Available Adapters](/website/docs/docs/available-adapters.md) page by following the steps given in [Documenting your adapter](/website/docs/docs/contributing/documenting-a-new-adapter.md).

Reviewer: The documenting your adapter link doesn't work in the preview?

Author: great catch -- fixed now!

> For any questions you may have, don't hesitate to ask in the [#adapter-ecosystem](https://getdbt.slack.com/archives/C030A0UF5LM) Slack channel. Likely there is someone there who has experienced a similar issue to what you are experiencing.

Reviewer: ```suggestion
For any questions you may have, don't hesitate to ask in the [#adapter-ecosystem](https://getdbt.slack.com/archives/C030A0UF5LM) Slack channel. The community is very helpful and likely has experienced a similar issue as you.
```
```suggestion
For any questions you may have, don't hesitate to ask in the [#adapter-ecosystem](https://getdbt.slack.com/archives/C030A0UF5LM) Slack channel. Likely there is someone there who has experienced a similar issue to what you are experiencing.
```

> For any questions you may have, don't hesitate to ask in the [#adapter-ecosystem](https://getdbt.slack.com/archives/C030A0UF5LM) Slack channel. Likely there is someone there who has experienced a similar issue to what you are experiencing.

Reviewer: ```suggestion
For any questions you may have, don't hesitate to ask in the [#adapter-ecosystem](https://getdbt.slack.com/archives/C030A0UF5LM) Slack channel. The community is very helpful and likely has experienced a similar issue as you.
```
```suggestion
For any questions you may have, don't hesitate to ask in the [#adapter-ecosystem](https://getdbt.slack.com/archives/C030A0UF5LM) Slack channel. Likely there is someone there who has experienced a similar issue to what you are experiencing.
```

2. In the middle: A set of *macros* responsible for generating SQL that is compliant with the target database.
3. (Optional) At the highest level: A set of *<Term id="materialization">materializations</Term>* that tell dbt how to turn model files into persisted objects in the database.

This guide will walk you through the first two steps, and provide some resources to help you validate that your new adapter is working correctly. Once the adapter is passing most of the functional tests (see ["Testing a new adapter"](testing-a-new-adapter)

> ), please let the community know that is available to use by adding the adapter to the [Available Adapters](/website/docs/docs/available-adapters.md) page by following the steps given in [Documenting your adapter](/website/docs/docs/contributing/documenting-a-new-adapter.md).

Author: ```suggestion
), please let the community know that is available to use by adding the adapter to the [Available Adapters](docs/available-adapters) page by following the steps given in [Documenting your adapter](docs/contributing/documenting-a-new-adapter).
```
